packages/next/src/server/lib/experimental/typed-env.ts

```diff
@@ -0,0 +1,30 @@
+import type { Env } from '@next/env'
+import { dirname, join } from 'node:path'
+import { mkdir, writeFile } from 'node:fs/promises'
+
+export async function createEnvDefinitions(distDir: string, env: Env) {
+  const envKeys = Object.keys(env ?? {})
```

Comment:
if there's no env and no existing file we can exist instead of still writing one